src/Resolution.ts

```diff
@@ -44,68 +55,116 @@ export default class Resolution {
+    if (provider.sendAsync === undefined) throw new ConfigurationError(ConfigurationErrorCode.IncorrectProvider);
+    return this.fromEip1193Provider({
+      request: (request: RequestArguments) =>
+        new Promise((resolve, reject) => {
+          provider.sendAsync(
+            { jsonrpc: '2.0', method: request.method, params: this.wrapArray(request.params), id: 1 },
+            (error: Error | null, result: JsonRpcResponse) => {
+              if (error) reject(error);
+              if (result.error) reject(new Error(result.error))
+              resolve(result.result);
+            },
+          );
+        }),
+    });
   }
 
   /**
-   * Creates a resolution instance from configured jsonRPCProvider
-   * @param provider - any jsonRPCprovider will work as long as it's prototype has send(method, params): Promise<any> method
+   * Create a resolution instance from web3 1.x version provider
+   * @param provider - an 1.x version provider from web3 ( must implement send(payload, callback) )
+   * @see https://github.com/ethereum/web3.js/blob/1.x/packages/web3-core-helpers/types/index.d.ts#L165
+   * @see https://github.com/ethereum/web3.js/blob/1.x/packages/web3-providers-http/src/index.js#L95
    */
-  static jsonRPCprovider(provider): Resolution {
-    return new this({ blockchain: { web3Provider: provider.send } });
+  static fromWeb3Version1Provider(provider: Web3Version1Provider): Resolution {
+    if (provider.send === undefined) throw new ConfigurationError(ConfigurationErrorCode.IncorrectProvider);
+    return this.fromEip1193Provider({
+      request: (request: RequestArguments) =>
+        new Promise((resolve, reject) => {
+          provider.send(
+            { jsonrpc: '2.0', method: request.method, params: this.wrapArray(request.params), id: 1 },
+            (error: Error | null, result: JsonRpcResponse) => {
+              if (error) reject(error);
+              if (result.error) reject(new Error(result.error))
+              resolve(result.result);
+            },
+          );
+        }),
+    });
+  }
+
+  /**
+   * Creates instance of resolution from provider that
+   * implements Ethers Provider#call interface
+   * @param provider - provider object
+   * @see https://github.com/ethers-io/ethers.js/blob/v4-legacy/providers/abstract-provider.d.ts#L91
+   * @see https://github.com/ethers-io/ethers.js/blob/v5.0.4/packages/abstract-provider/src.ts/index.ts#L224
+   * @see https://docs.ethers.io/ethers.js/v5-beta/api-providers.html#jsonrpcprovider-inherits-from-provider
+   * @see https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/json-rpc-provider.ts
+   */
+  static fromEthersProvider(provider: EthersProvider): Resolution {
+    if (provider.call === undefined) throw new ConfigurationError(ConfigurationErrorCode.IncorrectProvider);
+    return this.fromEip1193Provider({
+      request: async (request: RequestArguments) => await provider.call(request.params![0])
```

Comment:
I think it's worth to add a guard here checking a request method. Since this the code always executes `call` method, we should at least check that `request.method` equals `eth_call`, so we don't accidentally try to call another method and receive an unexpected API error.
```
  if (request.method !== 'eth_call') {
    throw new UnsupportedMethod() // something like that
  }
```